libraries/Microsoft.Bot.Builder.Ai/LocaleConverterMiddleware.cs

```diff
@@ -10,15 +10,17 @@ namespace Microsoft.Bot.Builder.Ai
     /// </summary>
     public class LocaleConverterMiddleware : IReceiveActivity
     {
-        private LocaleConverter localeConverter;
-        private string fromLocale;
-        private string toLocale;
+        private ILocaleConverter localeConverter; 
+        private readonly string toLocale;
+        private readonly Func<IBotContext, string> _getUserLocale;
+        private readonly Func<IBotContext, Task<bool>> _setUserLocale;
 
-        public LocaleConverterMiddleware(string fromLocale, string toLocale)
+        public LocaleConverterMiddleware(Func<IBotContext, string> getUserLocale, Func<IBotContext, Task<bool>> setUserLocale, string toLocale, ILocaleConverter localeConverter)
```

Comment:
I don't think this will quite work. 

Middleware is generally created as part of a Singleton. Certainly in ASP.Net we're promoting people to create the Middleware as part of Setup. 

The usage here is assuming the middleare is created for EACH CALL. That's not the case, as the Middleware will be re-used across multiple users, and multiple calls. 

The "per-turn" parameters (userLocale, etc) need to be done on each conversation turn, not as part of the creation of the Middleware. Certainly, the Middleware will support multiple users and the pattern here would break that model.